licensing/src/main/java/io/crate/license/LicenseMetaData.java

```diff
@@ -0,0 +1,183 @@
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.license;
+
+import io.crate.license.exception.LicenseMetadataParsingException;
+import org.elasticsearch.cluster.AbstractNamedDiffable;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Objects;
+
+public class LicenseMetaData extends AbstractNamedDiffable<MetaData.Custom> implements MetaData.Custom {
+
+    public static final String TYPE = "license";
+
+    private long expirationDateInMs;
+    private String issuedTo;
+    private String signature;
+
+    public LicenseMetaData(long expirationDateInMs, String issuedTo, String signature) {
+        this.expirationDateInMs = expirationDateInMs;
+        this.issuedTo = issuedTo;
+        this.signature = signature;
+    }
+
+    public LicenseMetaData(StreamInput in) throws IOException {
+        readFrom(in);
+    }
+
+    public long expirationDateInMs() {
+        return expirationDateInMs;
+    }
+
+    public String issuedTo() {
+        return issuedTo;
+    }
+
+    public String signature() {
+        return signature;
+    }
+
+    public void readFrom(StreamInput in) throws IOException {
+        this.expirationDateInMs = in.readLong();
+        this.issuedTo = in.readString();
+        this.signature = in.readString();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeLong(expirationDateInMs);
+        out.writeString(issuedTo);
+        out.writeString(signature);
+    }
+
+    /*
+     * LicenseMetaData XContent has the following structure:
+     *
+     * <pre>
+     *     {
+     *       "license": {
+     *           "expirationDateInMs": 202020202,
+     *           "issuedTo": "organisation",
+     *           "signture": "XXX"
+     *       }
+     *     }
+     * </pre>
+     */
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(TYPE);
+        builder.field("expirationDateInMs", expirationDateInMs);
```

Comment:
fyi: all the methods on the builder return the builder itself. So you could also write:

```
return builder
  .startObject(TYPE)
    .field(...)
    .field(...)
    .field(...)
  .endObject()
```

No need to change, just fyi.